### DISCUSSION TITLE
Feature Request: Possibility to force connection based on BSSID

### DISCUSSION THREAD

**Discussion by lealog** [joshschmelzle/lswifi#14] [Ideas]
Hi Josh,

Do you know if it's possible to force an STA to connect directly to a specific AP based on BSSID?
This could be an interesting feature to have on lswifi package.

For linux we can use WPA_supplicant or nmcli (ex. nmcli d wifi connect aa:bb:cc:dd:ee:gg password "mypassword").

Thanks :)

**Comment by joshschmelzle** [comment]
Performing connection actions with lswifi is out of scope for what I am willing to develop and test. Contributions are welcome though!

**Comment by lealog** [comment]
Thanks Josh. I know one software that is able to do that - https://www.nirsoft.net/utils/wifi_information_view.html
The command example is "WifiInfoView.exe /ConnectAP "MyNet" "00-25-9C-14-2C-39"".

If I found a way to implement by myself this "feature", I will send a PR :)

Thanks again.

**Comment by joshschmelzle** [comment]
Yes, it is possible to do what nirsoft is doing by using the wlanapi.h header. However, I did not implement the corresponding wrapper code. I think the first step would be adding corresponding wrapper code to `wlanapi.py`. Adding the new args to `appsetup.py` is straight forward. I think then the connection code (which calls the wrapper code) to `client.py`, and probably refactor `__main__.py` to call the client connection code instead of scanning for nearby networks.

**Comment by joshschmelzle** [comment]
I've recently come up with a need to do something like this where it would be ideal to have the ability to select a specific BSS within the ESS of multiple nearby APs for testing.
